File: nl.naturalis.nda.domain/src/main/java/nl/naturalis/nda/domain/GatheringEvent.java
package nl.naturalis.nda.domain;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class GatheringEvent {

	private String projectTitle;
	private String worldRegion;
	private String continent;
	private String country;
	private String iso3166Code;
	private String provinceState;
	private String island;
	private String locality;
	private String city;
	private String sublocality;
	private String localityText;
	private Date dateTimeBegin;
	private Date dateTimeEnd;
	private String method;
	private String altitude;
	private String altitudeUnifOfMeasurement;
	private String depth;
	private String depthUnitOfMeasurement;
	private List<Agent> gatheringAgents;
	private List<GatheringSiteCoordinates> siteCoordinates;
    private List<ChronoStratigraphy> chronoStratigraphy;
    private List<BioStratigraphy> bioStratigraphic;
    private List<LithoStratigraphy> lithoStratigraphy;

	public void addGatheringAgent(Agent agent)
	{
		if (gatheringAgents == null) {
			gatheringAgents = new ArrayList<Agent>();
		}
		gatheringAgents.add(agent);
	}


	public void addSiteCoordinates(GatheringSiteCoordinates coordinates)
	{
		if (siteCoordinates == null) {
			siteCoordinates = new ArrayList<GatheringSiteCoordinates>();
		}
		siteCoordinates.add(coordinates);
	}


	public void addSiteCoordinates(Double latitude, Double longitude)
	{
		addSiteCoordinates(new GatheringSiteCoordinates(latitude, longitude));
	}


	public String getProjectTitle()
	{
		return projectTitle;
	}


	public void setProjectTitle(String projectTitle)
	{
		this.projectTitle = projectTitle;
	}


	public String getWorldRegion()
	{
		return worldRegion;
	}


	public void setWorldRegion(String worldRegion)
	{
		this.worldRegion = worldRegion;
	}


	/**
	 * N.B. This is not strictly ABCD, but this information is provided by some
	 * Naturalis data sources, and the meaning and specifity of continent is
	 * rather more obvious than world region. If a data source provides a
	 * continent, but not a world region, {@code content} and
	 * {@code worldRegion} will both be set to the provided continent. If a data
	 * source provides a world region, but not a continent, only the
	 * {@code worldRegion} field will be set to the world region, unless a
	 * continent could be parsed out of the location data while populating the
	 * search index.
	 * 
	 * @return
	 */
	public String getContinent()
	{
		return continent;
	}


	public void setContinent(String continent)
	{
		this.continent = continent;
	}


	public String getCountry()
	{
		return country;
	}


	public void setCountry(String country)
	{
		this.country = country;
	}


	/**
	 * Get the ISO3166-1 or ISO3166-3 country code for the
	 * {@code GatheringEvent}.
	 * 
	 * @see http://wiki.tdwg.org/twiki/bin/view/ABCD/AbcdConcept0962
	 * 
	 * @return The ISO3166-1 or ISO3166-3 country code for the
	 *         {@code GatheringEvent}
	 */
	public String getIso3166Code()
	{
		return iso3166Code;
	}


	public void setIso3166Code(String iso3166Code)
	{
		this.iso3166Code = iso3166Code;
	}


	public String getProvinceState()
	{
		return provinceState;
	}


	public void setProvinceState(String provinceState)
	{
		this.provinceState = provinceState;
	}


	public String getIsland()
	{
		return island;
	}


	public void setIsland(String island)
	{
		this.island = island;
	}


	public String getLocality()
	{
		return locality;
	}


	public void setLocality(String locality)
	{
		this.locality = locality;
	}


	/**
	 * N.B. This is not strictly ABCD, but it enables the generation of IPTC
	 * data ({@link Iptc4xmpExt} objects) from {@code GatheringEvent}s. If a
	 * Naturalis data source happens to provide a city but not a locality, both
	 * {@code city} and {@code locality} will be set to the provided locality.
	 * If a locality was provided but not a city, only the {@code locality}
	 * field will be set, unless a city could be parsed out of the location data
	 * while populating the search index.
	 * 
	 * @return
	 */

	public String getCity()
	{
		return city;
	}


	public void setCity(String city)
	{
		this.city = city;
	}


	/**
	 * N.B. This is not strictly ABCD, but it enables the generation of IPTC
	 * data ({@link Iptc4xmpExt} objects) from {@code GatheringEvent}s.
	 * 
	 * @return
	 */
	public String getSublocality()
	{
		return sublocality;
	}


	public void setSublocality(String sublocality)
	{
		this.sublocality = sublocality;
	}


	public String getLocalityText()
	{
		return localityText;
	}


	public void setLocalityText(String localityText)
	{
		this.localityText = localityText;
	}


	public Date getDateTimeBegin()
	{
		return dateTimeBegin;
	}


	public void setDateTimeBegin(Date dateTimeBegin)
	{
		this.dateTimeBegin = dateTimeBegin;
	}


	public Date getDateTimeEnd()
	{
		return dateTimeEnd;
	}


	public void setDateTimeEnd(Date dateTimeEnd)
	{
		this.dateTimeEnd = dateTimeEnd;
	}


	public String getMethod()
	{
		return method;
	}


	public void setMethod(String method)
	{
		this.method = method;
	}


	public String getAltitude()
	{
		return altitude;
	}


	public void setAltitude(String altitude)
	{
		this.altitude = altitude;
	}


	public String getAltitudeUnifOfMeasurement()
	{
		return altitudeUnifOfMeasurement;
	}


	public void setAltitudeUnifOfMeasurement(String altitudeUnifOfMeasurement)
	{
		this.altitudeUnifOfMeasurement = altitudeUnifOfMeasurement;
	}


	public String getDepth()
	{
		return depth;
	}


	public void setDepth(String depth)
	{
		this.depth = depth;
	}


	public String getDepthUnitOfMeasurement()
	{
		return depthUnitOfMeasurement;
	}


	public void setDepthUnitOfMeasurement(String depthUnitOfMeasurement)
	{
		this.depthUnitOfMeasurement = depthUnitOfMeasurement;
	}


	public List<Agent> getGatheringAgents()
	{
		return gatheringAgents;
	}


	public void setGatheringAgents(List<Agent> gatheringAgents)
	{
		this.gatheringAgents = gatheringAgents;
	}


	public List<GatheringSiteCoordinates> getSiteCoordinates()
	{
		return siteCoordinates;
	}


	public void setSiteCoordinates(List<GatheringSiteCoordinates> siteCoordinates)
	{
		this.siteCoordinates = siteCoordinates;
	}


	public List<ChronoStratigraphy> getChronoStratigraphy()
	{
		return chronoStratigraphy;
	}


	public void setChronoStratigraphy(List<ChronoStratigraphy> chronoStratigraphy)
	{
		this.chronoStratigraphy = chronoStratigraphy;
	}


	public List<BioStratigraphy> getBiostratigraphic()
	{
		return bioStratigraphic;
	}


	public void setBiostratigraphic(List<BioStratigraphy> bioStratigraphic)
	{
		this.bioStratigraphic = bioStratigraphic;
	}


	public List<LithoStratigraphy> getLithoStratigraphy()
	{
		return lithoStratigraphy;
	}


	public void setLithoStratigraphy(List<LithoStratigraphy> lithoStratigraphy)
	{
		this.lithoStratigraphy = lithoStratigraphy;
	}

}

File: nl.naturalis.nba.client.php/model/GatheringEvent.php
<?php

namespace nl\naturalis\nba\client\php\model;

use \DateTime;

class GatheringEvent extends NBADomainObject {

	private $projectTitle;
	private $worldRegion;
	private $continent;
	private $country;
	private $iso3166Code;
	private $provinceState;
	private $island;
	private $locality;
	private $city;
	private $sublocality;
	private $localityText;
	private $dateTimeBegin;
	private $dateTimeEnd;
	private $method;
	private $altitude;
	private $altitudeUnifOfMeasurement;
	private $depth;
	private $depthUnitOfMeasurement;
	private $gatheringAgents;
	private $siteCoordinates;
	private $chronoStratigraphy;
	private $bioStratigraphic;
	private $lithoStratigraphy;

	public function addGatheringAgent(Agent $agent)
	{
		if (gatheringAgents == null) {
			$this->gatheringAgents = array();
		}
		$this->gatheringAgents[] = $agent;
	}

	public function addSiteCoordinates(GatheringSiteCoordinates $coordinates)
	{
		if (siteCoordinates == null) {
			$this->siteCoordinates = array();
		}
		$this->siteCoordinates[] = $coordinates;
	}

	public function addSiteCoordinates(float $latitude, float $longitude)
	{
		$this->addSiteCoordinates(new GatheringSiteCoordinates(latitude, longitude));
	}

	public function getProjectTitle()
	{
		return $this->projectTitle;
	}

	public function setProjectTitle(string $projectTitle)
	{
		$this->projectTitle = $projectTitle;
	}

	public function getWorldRegion()
	{
		return $this->worldRegion;
	}

	public function setWorldRegion(string $worldRegion)
	{
		$this->worldRegion = $worldRegion;
	}

	public function getContinent()
	{
		return $this->continent;
	}

	public function setContinent(string $continent)
	{
		$this->continent = $continent;
	}

	public function getCountry()
	{
		return $this->country;
	}

	public function setCountry(string $country)
	{
		$this->country = $country;
	}

	public function getIso3166Code()
	{
		return $this->iso3166Code;
	}

	public function setIso3166Code(string $iso3166Code)
	{
		$this->iso3166Code = $iso3166Code;
	}

	public function getProvinceState()
	{
		return $this->provinceState;
	}

	public function setProvinceState(string $provinceState)
	{
		$this->provinceState = $provinceState;
	}

	public function getIsland()
	{
		return $this->island;
	}

	public function setIsland(string $island)
	{
		$this->island = $island;
	}

	public function getLocality()
	{
		return $this->locality;
	}

	public function setLocality(string $locality)
	{
		$this->locality = $locality;
	}

	public function getCity()
	{
		return $this->city;
	}

	public function setCity(string $city)
	{
		$this->city = $city;
	}

	public function getSublocality()
	{
		return $this->sublocality;
	}

	public function setSublocality(string $sublocality)
	{
		$this->sublocality = $sublocality;
	}

	public function getLocalityText()
	{
		return $this->localityText;
	}

	public function setLocalityText(string $localityText)
	{
		$this->localityText = $localityText;
	}

	public function getDateTimeBegin()
	{
		return $this->dateTimeBegin;
	}

	public function setDateTimeBegin(DateTime $dateTimeBegin)

	{
		$this->dateTimeBegin = $dateTimeBegin;
	}

	public function getDateTimeEnd()
	{
		return $this->dateTimeEnd;
	}

	public function setDateTimeEnd(DateTime $dateTimeEnd)
	{
		$this->dateTimeEnd = $dateTimeEnd;
	}

	public function getMethod()
	{
		return $this->method;
	}

	public function setMethod(string $method)
	{
		$this->method = $method;
	}

	public function getAltitude()
	{
		return $this->altitude;
	}

	public function setAltitude(string $altitude)
	{
		$this->altitude = $altitude;
	}

	public function getAltitudeUnifOfMeasurement()
	{
		return $this->altitudeUnifOfMeasurement;
	}

	public function setAltitudeUnifOfMeasurement(string $altitudeUnifOfMeasurement)
	{
		$this->altitudeUnifOfMeasurement = $altitudeUnifOfMeasurement;
	}

	public function getDepth()
	{
		return $this->depth;
	}

	public function setDepth(string $depth)
	{
		$this->depth = $depth;
	}

	public function getDepthUnitOfMeasurement()
	{
		return $this->depthUnitOfMeasurement;
	}

	public function setDepthUnitOfMeasurement(string $depthUnitOfMeasurement)
	{
		$this->depthUnitOfMeasurement = $depthUnitOfMeasurement;
	}

	public function getGatheringAgents()
	{
		return $this->gatheringAgents;
	}

	public function setGatheringAgents(array $gatheringAgents)
	{
		$this->gatheringAgents = $gatheringAgents;
	}

	public function getSiteCoordinates()
	{
		return $this->siteCoordinates;
	}

	public function setSiteCoordinates(array $siteCoordinates)
	{
		$this->siteCoordinates = $siteCoordinates;
	}

	public function getChronoStratigraphy()
	{
		return $this->chronoStratigraphy;
	}

	public function setChronoStratigraphy($chronoStratigraphy)
	{
		$this->chronoStratigraphy = $chronoStratigraphy;
	}

	public function getBiostratigraphic()
	{
		return $this->bioStratigraphic;
	}

	public function setBiostratigraphic($bioStratigraphic)
	{
		$this->bioStratigraphic = $bioStratigraphic;
	}

	public function getLithoStratigraphy()
	{
		return $this->lithoStratigraphy;
	}

	public function setLithoStratigraphy($lithoStratigraphy)
	{
		$this->lithoStratigraphy = $lithoStratigraphy;
	}
}
